config/kdump.py

```diff
@@ -1,44 +1,90 @@
-from swsscommon.swsscommon import ConfigDBConnector
+from utilities_common.cli import AbbreviationGroup, pass_db
+
 
-@click.group(cls=clicommon.AbbreviationGroup, name="kdump")
+#
+# 'kdump' group ('sudo config kdump ...')
+#
+@click.group(cls=AbbreviationGroup, name="kdump")
 def kdump():
-    """ Configure kdump """
-    if os.geteuid() != 0:
-        exit("Root privileges are required for this operation")
-
-@kdump.command()
-def disable():
-    """Disable kdump operation"""
-    config_db = ConfigDBConnector()
-    if config_db is not None:
-        config_db.connect()
-        config_db.mod_entry("KDUMP", "config", {"enabled": "false"})
-
-@kdump.command()
-def enable():
-    """Enable kdump operation"""
-    config_db = ConfigDBConnector()
-    if config_db is not None:
-        config_db.connect()
-        config_db.mod_entry("KDUMP", "config", {"enabled": "true"})
-
-@kdump.command()
+    """Configure the KDUMP mechanism"""
+    pass
+
+#
+# 'disable' command ('sudo config kdump disable')
+#
+@kdump.command(name="disable", short_help="Disable the KDUMP mechanism")
+@pass_db
+def kdump_disable(db):
+    """Disable the KDUMP mechanism"""
+    kdump_table = db.cfgdb.get_table("KDUMP")
+    if not kdump_table:
+        click.echo("Unable to retrieve 'KDUMP' table from Config DB.")
+        sys.exit(1)
+
+    if "config" not in kdump_table:
+        click.echo("Unable to retrieve key 'config' from KDUMP table.")
+        sys.exit(2)
+
+    db.cfgdb.mod_entry("KDUMP", "config", {"enabled": "false"})
+
+
+#
+# 'enable' command ('sudo config kdump enable')
+#
+@kdump.command(name="enable", short_help="Enable the KDUMP mechanism")
+@pass_db
+def kdump_enable(db):
+    """Enable the KDUMP mechanism"""
+    kdump_table = db.cfgdb.get_table("KDUMP")
+    if not kdump_table:
+        click.echo("Unable to retrieve KDUMP table from CONFIG DB.")
+        sys.exit(3)
+
+    if "config" not in kdump_table:
+        click.echo("Unable to retrieve key 'config' from KDUMP table.")
+        sys.exit(4)
+
+    db.cfgdb.mod_entry("KDUMP", "config", {"enabled": "true"})
+
+
+#
+# 'memory' command ('sudo config kdump memory ...')
+#
+@kdump.command(name="memory", short_help="Configure the memory for KDUMP mechanism")
 @click.argument('kdump_memory', metavar='<kdump_memory>', required=True)
-def memory(kdump_memory):
-    """Set memory allocated for kdump capture kernel"""
-    config_db = ConfigDBConnector()
-    if config_db is not None:
-        config_db.connect()
-        config_db.mod_entry("KDUMP", "config", {"memory": kdump_memory})
-
-@kdump.command('num-dumps')
+@pass_db
+def kdump_memory(db, kdump_memory):
+    """Reserve memory for kdump capture kernel"""
+    kdump_table = db.cfgdb.get_table("KDUMP")
+    if not kdump_table:
+        click.echo("Unable to retrieve KDUMP table from CONFIG DB.")
+        sys.exit(5)
+
+    if "config" not in kdump_table:
+        click.echo("Unable to retrieve key 'config' from KDUMP table.")
+        sys.exit(6)
```

Comment:
Could you reuse code? I see it 4 times in this file.

Reply:
Updated!